zeroex/orderwatch/order_watcher.go

```diff
@@ -288,6 +292,20 @@ func (w *Watcher) mainLoop(ctx context.Context) error {
 	}
 }
 
+func drainBlockEventsChan(blockEventsChan chan []*blockwatch.Event) []*blockwatch.Event {
+	allEvents := []*blockwatch.Event{}
+L:
+	for {
+		select {
+		case moreEvents := <-blockEventsChan:
+			allEvents = append(allEvents, moreEvents...)
```

Comment:
Can we add a maximum size for `allEvents` here? Infinite loops make me nervous. There's a slight chance that new events could come in before we have the time to process them, especially on testnets with a higher rate of mining new blocks or when CPU usage is high. It's not something that strikes me as super likely, but the fix is also really easy.

```suggestion
			allEvents = append(allEvents, moreEvents...)
			if len(allEvents) >= maxBlockEventsToHandle {
			    break L
			}
```